File: 0213_gmall/gmall-canal/src/main/scala/com/atguigu/canal/MyKafkaSender.scala
package com.atguigu.canal

import org.apache.kafka.clients.producer.{KafkaProducer, ProducerConfig, ProducerRecord}
import org.apache.kafka.common.serialization.StringSerializer
import java.util.Properties


object MyKafkaSender {
  var kafkaProducer:KafkaProducer[String, String]=_
  def createrKafkaProducer: KafkaProducer[String, String] = {

    val properties = new Properties
    properties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "hadoop202:9092,hadoop203:9092,hadoop204:9092")
    properties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, classOf[StringSerializer])
    properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, classOf[StringSerializer])
    val producer = new KafkaProducer[String, String](properties)
    producer
  }

  def send(topic: String, msg: String): Unit = {
    if (kafkaProducer == null) kafkaProducer = createrKafkaProducer
    kafkaProducer.send(new ProducerRecord[String, String](topic, msg))
  }
}

File: 0213_gmall/gmall-canal/src/main/java/com/atguigui/canal/MyKafkaSender.java
package com.atguigui.canal;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerConfig;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.apache.kafka.common.serialization.StringSerializer;

import java.util.Properties;

public class MyKafkaSender {

    private  static KafkaProducer<String ,String> kafkaProducer=null;

    public static KafkaProducer<String,String> createrKafkaProducer(){
        Properties properties = new Properties();
        properties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG,"hadoop202:9092,hadoop203:9092,hadoop204:9092");
        properties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
        properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,StringSerializer.class);

        KafkaProducer<String, String> producer =null;
        try {
            producer= new KafkaProducer<>(properties);
        } catch (Exception e) {
            e.printStackTrace();
        }

        return producer;
    }
    public static void send(String topic,String msg){
        if(kafkaProducer==null){
            kafkaProducer=createrKafkaProducer();
        }
        kafkaProducer.send(new ProducerRecord<>(topic,msg));
    }
}
